buildSrc/src/main/kotlin/targetsJvm.kt

```diff
@@ -0,0 +1,29 @@
+import org.gradle.jvm.toolchain.*
+import org.gradle.kotlin.dsl.*
+import org.jetbrains.kotlin.gradle.dsl.*
+
+fun KotlinMultiplatformExtension.configureJvm(block: TargetBuilder.() -> Unit = {}) {
+    jvmToolchain {
+        this as JavaToolchainSpec
+        languageVersion.set(JavaLanguageVersion.of(8))
+    }
+    jvm {
+        val latestTest by testRuns.creating {
+            executionTask.configure {
+                javaLauncher.set(
+                    project.extensions.getByType<JavaToolchainService>().launcherFor {
+                        languageVersion.set(JavaLanguageVersion.of(17))
+                    }
+                )
+            }
+        }
+        testRuns.all {
+            executionTask.configure {
+                // ActiveProcessorCount is used here, to make sure local setup is similar as on CI
+                // Github Actions linux runners have 2 cores
+                jvmArgs("-Xmx4g", "-XX:ActiveProcessorCount=2")
```

Comment:
Are there JDK builds you could use that would get this right?

Reply:
not sure, that I understand, what you mean :)

Reply:
Never mind, it's the reverse.  I thought JDK should pick up the ActiveProcessorCount correctly even for shared compute environments.  That's the point, it does, this makes it work predictably the same even on a local powerful dev host.